### PR TITLE
make sure that MArray are noted as Local AS

### PR DESCRIPTION
**Pull request by vchuravy**
@lcw maybe this helps with your perf problems (didn't have a chance to test the implementation yet so likely buggy)

### CODE REVIEW

**Comment by vchuravy** [comment]
On second thought that is actually unlikely to help much since all these allocations have already been promoted from mem to reg...

**Comment by lcw** [comment]
This seems to speed up https://github.com/lcw/Heptapus.jl/tree/e9da3ed68c473eb4d4a28b6ceb3843d686aec36d/examples/volumerhs-small by a factor of 2 to 3!  Still not quite the same as clang but now within a factor of ~2!

**Review by lcw — COMMENTED** [review]
I am not sure exactly how to test this but it speeds up my kernel that use StaticArrays.  Do we need to also add similar overdubs for `SArray`?

**Comment by vchuravy** [comment]
Looking into what changed when we turned this on in the ptx we now see:

```
mov.u64     %SPL, __local_depot0; 
add.u64     %rd12, %SPL, 0; 
mov.f32     %f3, 0f00000000;
st.local.v4.f32     [%rd12], {%f3, %f3, %f3, %f3};
```

Which is actually allocating "stack" memory, this is further confirmed by the `opt.ll` having `alloca i160` which equals `20 bytes` or `5 F32`.
Again rather counter-intuitive...

**Comment by vchuravy** [comment]
Confirmed by.

Before: 
```
ptxas info    : 0 bytes gmem
ptxas info    : Compiling entry function 'ptxcall_volumerhs__34' for 'sm_70'
ptxas info    : Function properties for ptxcall_volumerhs__34
    8 bytes stack frame, 4 bytes spill stores, 4 bytes spill loads
ptxas info    : Used 255 registers, 1124 bytes smem, 536 bytes cmem[0], 8 bytes cmem[2]
```

After:
```
ptxas info    : 0 bytes gmem
ptxas info    : Compiling entry function 'ptxcall_volumerhs__8' for 'sm_70'
ptxas info    : Function properties for ptxcall_volumerhs__8
    160 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 236 registers, 1124 bytes smem, 536 bytes cmem[0], 8 bytes cmem[2]
```

Note the 160 bytes stack-frame.

**Comment by lcw** [comment]
I see, and for reference clang doesn't use a stack-frame

```
ptxas info    : 0 bytes gmem, 296 bytes cmem[3]
ptxas info    : Compiling entry function '_Z9volumerhsILi5ELi125ELi5EEvPfPKfS2_fS2_i' for 'sm_70'
ptxas info    : Function properties for _Z9volumerhsILi5ELi125ELi5EEvPfPKfS2_fS2_i
    0 bytes stack frame, 0 bytes spill stores, 0 bytes spill loads
ptxas info    : Used 80 registers, 1100 bytes smem, 396 bytes cmem[0]
```
